azure-proto-core/Adapters/ModelAdapters.cs

```diff
@@ -24,4 +24,22 @@ protected TrackedResource(ResourceIdentifier id, Location location, T data)
             return other.Model;
         }
     }
+
+    //Or call generic resource, other resource??
```

Comment:
For each of these classes with Type arguments, we should consider having a base class without type arguments (i.e. ChildResource)  This allows us to write code to address all SubResources as a class (for example, in method signatures).  This is a pattern often followed in the .net framework (for example, IList, IList<T>).